lib/Doctrine/ORM/Mapping/ReflectionEnumProperty.php

```diff
@@ -67,18 +75,37 @@ public function setValue($object, $value = null): void
     {
         if ($value !== null) {
             $enumType = $this->enumType;
-            try {
-                $value = $enumType::from($value);
-            } catch (ValueError $e) {
-                assert(is_string($value) || is_int($value));
-
-                throw MappingException::invalidEnumValue(
-                    get_class($object),
-                    $this->originalReflectionProperty->getName(),
-                    (string) $value,
-                    $enumType,
-                    $e
-                );
+
+            if (is_array($value)) {
```

Comment:
What do you think of refactoring it to have less duplication?

Yo could introduce a private helper method `initializeEnumValue` that extracts what is currently in line 81-94 and 96-108. I believe its exactly the same code or did I miss something?

Reply:
The code is exactly the same yeah. I've added the helper in [01a1b75](https://github.com/doctrine/orm/pull/9497/commits/01a1b75086982bed44c1be46c40aeca1f613b774).